src/libopensc/card.c

```diff
@@ -397,41 +397,25 @@ int sc_lock(sc_card_t *card)
 		return r;
 	if (card->lock_count == 0) {
 		if (card->reader->ops->lock != NULL) {
-			r = card->reader->ops->lock(card->reader);
-			while (r == SC_ERROR_CARD_RESET || r == SC_ERROR_READER_REATTACHED) {
+			r = card->reader->ops->lock(card->reader, 0);
```

Comment:
@dengert:
> Why did you change this?

Since the card is going to be reinitialized anyway on reset there is (currently)
no point in calling card_reader_lock_obtained or reopening sm channel.